File: 35. Search Insert Position/35. Search Insert Position.cpp
class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int i = 0, j = nums.size() - 1;
        while (i < j)
        {
            int mid = (i + j) / 2;
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                j = mid - 1;
            else
                i = mid + 1;
        }
        if (nums[i] >= target)
            return i;
        else
            return i + 1;
    }
};

File: 35. Search Insert Position/35. Search Insert Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int start = 0, end = nums.length;
        while(start < end) {
            int mid = start + (end - start)/2;
            if(nums[mid] > target) {
                end = mid;
            } else if(nums[mid] < target) {
                start = mid + 1;
            } else {
                return mid;
            }
        }
        return start;
    }
}